src/us/kbase/workspace/kbase/InitWorkspaceServer.java

```diff
@@ -232,49 +233,133 @@ private static AdministratorHandler getAdminHandler(
+					new TypedObjectValidator(typeProvider),
+					tfm,
+					loadListeners(cfg));
+		} catch (WorkspaceCommunicationException e) { // this is really hard to test
+			throw new WorkspaceInitException(e.getMessage(), e);
+		}
+		final IdReferenceHandlerSetFactoryBuilder builder = IdReferenceHandlerSetFactoryBuilder
+				.getBuilder(maxUniqueIdCountPerCall)
+				.withFactory(new HandleIdHandlerFactory(hsc))
+				.withFactory(getShockIdHandlerFactory(cfg, auth))
+				.withFactory(getSampleIdHandlerFactory(cfg, auth, rep))
+				.build();
+		final WorkspaceServerMethods wsmeth = new WorkspaceServerMethods(ws, builder, auth);
+		final WorkspaceAdministration.Builder adminbuilder = WorkspaceAdministration.getBuilder(
+				getAdminHandler(cfg, ws), (user, token) -> wsmeth.validateUser(user, token));
+		if (typeDelegator.isPresent()) {
+			AdministrationCommandSetInstaller.install(
+					adminbuilder, wsmeth, typeDelegator.get().delegator);
+		} else {
+			AdministrationCommandSetInstaller.install(
+					adminbuilder, wsmeth, (LocalTypeServerMethods) types);
+		}
+		return new WorkspaceInitResults(wsmeth, adminbuilder.build(), types);
+	}
+	
+	private static class TypeDelegation {
+		private final WorkspaceClient delegationClient;
+		private final WorkspaceDelegator delegator;
+
+		public TypeDelegation(
+				final WorkspaceClient delegationClient,
+				final WorkspaceDelegator delegator) {
+			this.delegationClient = delegationClient;
+			this.delegator = delegator;
+		}
+	}
+
+	private static Optional<TypeDelegation> getTypeDelegator(
+			final KBaseWorkspaceConfig cfg,
+			final InitReporter rep) // DO NOT use the rep to report failures. Throw instead.
+			throws WorkspaceInitException {
+		if (cfg.getTypeDelegationTarget() == null) {
+			return Optional.empty();
+		}
+		final WorkspaceClient cli = new WorkspaceClient(cfg.getTypeDelegationTarget());
+		try {
+			cli.ver();
+		} catch (IOException | JsonClientException e) {
+			// this code was tested manually. If you make changes retest
+			throw new WorkspaceInitException(
+					"Failed contacting type delegation workspace: " + e.getMessage(), e);
+		}
+		if (cfg.getTypeDelegationTarget().getProtocol().equals("http")) {
+			rep.reportInfo("Warning - the Type Delegation url uses insecure http. " +
+					"https is recommended.");
+			cli.setIsInsecureHttpConnectionAllowed(true);
+		}
+		return Optional.of(new TypeDelegation(
+				cli,
+				new WorkspaceDelegator(
+						cfg.getTypeDelegationTarget(),
+						new WorkspaceDelegator.WorkspaceClientProvider() {
+			
+							@Override
+							public WorkspaceClient getClient(
+									final URL workspaceURL,
+									final AuthToken token)
+									throws UnauthorizedException, IOException {
+								return new WorkspaceClient(workspaceURL, token);
+							}
+							
+							@Override
+							public WorkspaceClient getClient(final URL workspaceURL)
+									throws UnauthorizedException, IOException {
+								return cli;
```

Comment:
Once I add integration tests for methods that don't take tokens this line'll be covered